crates/extensions/tedge_config_manager/src/lib.rs

```diff
@@ -147,27 +155,111 @@ impl ConfigManagerBuilder {
     fn watched_directory(config: &ConfigManagerConfig) -> PathBuf {
         config.plugin_config_dir.clone()
     }
+
+    /// Extract a config actor request from an MQTT message
+    fn mqtt_message_parser(
+        config: &ConfigManagerConfig,
+    ) -> impl Fn(MqttMessage) -> Option<ConfigInput> {
+        let config = config.clone();
+        move |message| match ConfigOperation::request_from_message(&config, &message) {
+            Ok(Some(request)) => Some(request.into()),
+            Ok(None) => None,
+            Err(err) => {
+                error!("Received invalid config request: {err}");
+                None
+            }
+        }
+    }
 }
 
 impl RuntimeRequestSink for ConfigManagerBuilder {
     fn get_signal_sender(&self) -> DynSender<RuntimeRequest> {
-        Box::new(self.signal_sender.clone())
+        self.box_builder.get_signal_sender()
     }
 }
 
 impl Builder<ConfigManagerActor> for ConfigManagerBuilder {
     type Error = LinkError;
 
     fn try_build(self) -> Result<ConfigManagerActor, Self::Error> {
-        let mqtt_publisher = LoggingSender::new("Tedge-Config-Manager".into(), self.mqtt_publisher);
+        let (output_sender, input_receiver) = self.box_builder.build().into_split();
 
         Ok(ConfigManagerActor::new(
             self.config,
             self.plugin_config,
-            self.receiver,
-            mqtt_publisher,
+            input_receiver,
+            output_sender,
             self.download_sender,
             self.upload_sender,
         ))
     }
 }
+
+impl MessageSource<GenericCommandData, NoConfig> for ConfigManagerBuilder {
+    fn connect_sink(&mut self, config: NoConfig, peer: &impl MessageSink<GenericCommandData>) {
+        self.box_builder.connect_mapped_sink(
+            config,
+            &peer.get_sender(),
+            |data: ConfigOperationData| match data {
+                ConfigOperationData::State(ConfigOperation::Snapshot(topic, payload)) => Some(
+                    GenericCommandState::new(topic, payload.status.to_string(), payload.to_value())
+                        .into(),
+                ),
+                ConfigOperationData::State(ConfigOperation::Update(topic, payload)) => Some(
+                    GenericCommandState::new(topic, payload.status.to_string(), payload.to_value())
+                        .into(),
+                ),
+                ConfigOperationData::Metadata { topic, types } => {
+                    let Some(operation) = MqttSchema::get_operation_name(topic.as_ref()) else {
+                        return None;
+                    };
+                    Some(GenericCommandData::Metadata(GenericCommandMetadata {
+                        operation,
+                        payload: json!( {
+                            "types": types
+                        }),
+                    }))
+                }
+            },
+        )
+    }
+}
```

Comment:
I not happy with this connection logic.

This is unfortunate but each operation actor is using is own representation for command metadata and states. Here Payload are used. The log actor is using Command<Payload>. I will try to make this more consistent once the main feature properly working.